src/App.vue

```diff
@@ -1,23 +1,103 @@
+
+
 <template>
   <div id="app">
-    <img src="./assets/logo.png">
-    <router-view/>
+    <toolbar></toolbar>
+    <router-view :meals="meals" @mealUsed="markUsed"/>
   </div>
 </template>
 
 <script>
+  import { map, pipe, curry, trace } from "@/utils/fp";
+  import MealService from "@/services/meals";
+  import mediator from "@/mediator";
+
+  import Toolbar from "@/components/Toolbar.vue"
+
+  const identifyMeal = (meal, id) => Object.assign({id}, meal);
+
+  const unIdentifyMeal = function(meal) {
+  	let unidentifiedMeal = {};
+
+    for(let prop in meal) {
+    	if(meal.hasOwnProperty(prop) && prop !== "id") {
+    		unidentifiedMeal[prop] = meal[prop];
+      }
+    }
+    return unidentifiedMeal
+  }
+
+  const updateMeal = curry((changes, meal) => Object.assign(meal, changes));
+
+  const changeAtIndex = curry(function(fx, targetIdx, target, currentIdx) {
+  	if(target.id === targetIdx) {
+  		return fx(target);
+    } else {
+  		return target
+    }
+  });
+
 export default {
+  data: function() {
+    return {
+    	meals: []
+    }
+  },
+  created: function() {
+    mediator.$on("mealUsed", this.markUsed);
+    mediator.$on("saveMeal", this.saveMeal)
+    mediator.$on("sorted", (meals) => this.meals = meals)
+  },
+  mounted: function() {
+    MealService.get()
```

Comment:
lookit, we can use services.

But really, a key point here is that only this main, root component is responsible for getting/setting data from outside sources. There are probably going to be times when we want fully encapsulated components that understand out to get and set their own data, but my understanding is that it's a bit of an anti-pattern to have each component in a tree get it's own data, and I can see why.